src/chigame/users/views.py

```diff
@@ -87,14 +87,16 @@ def user_profile_detail_view(request, pk):
         profile = get_object_or_404(UserProfile, user__pk=pk)
         if request.user.pk == pk:
             return render(request, "users/userprofile_detail.html", {"object": profile})
-        is_friend = profile.friends.filter(pk=request.user.pk).exists()
+        is_friend = None
         friendship_request = None
-        if not is_friend:
-            curr_user = User.objects.get(pk=request.user.id)
-            other_user = profile.user
-            friendship_request = FriendInvitation.objects.filter(
-                Q(sender=curr_user, receiver=other_user) | Q(sender=other_user, receiver=curr_user)
-            ).first()
+        if request.user.pk:
+            profile.friends.filter(pk=request.user.pk).exists()
```

Comment:
You would need to set this line equal to is_friend. Currently, even if a friend request is accepted, the profile page is still stuck on the accept or decline page.